### PR TITLE
Forward-port some 1.x.x stuff we want in here, remove deprecations, & replace db with ledger everywhere appropriate

### PR DESCRIPTION
**Pull request by cap10morgan**
Counterpart to https://github.com/fluree/db/pull/170

### CODE REVIEW

**Comment by cap10morgan** [comment]
With this and https://github.com/fluree/db/pull/170 merged, we should be good to roll 2.0.0-alpha1

File: src/fluree/db/ledger/indexing.clj [hunk 1]
```diff
@@ -54,13 +54,13 @@
     []))
 
 (defn rebalance-leaf
-  "Splits leaf nodes if the combined size of it's flakes is greater than
+  "Splits leaf nodes if the combined size of its flakes is greater than
```
Comment (zonotope) [inline src/fluree/db/ledger/indexing.clj:57]: Yeah, that was definitely me 😜

File: src/fluree/db/peer/http_api.clj [hunk 2]
```diff
@@ -794,34 +793,36 @@
...
-                        db-name          (keyword network db)
-                        _                (when-not (and network db type)
-                                           (throw (ex-info (str "Incomplete request. At least a network, db and type are required. Provided network: " network " db: " db " type: " type " key: " key)
-                                                           {:status 400 :error :db/invalid-request})))
+                        {:keys [network ledger type key]} params
+                        ledger-name      (keyword network ledger)
+                        _                (when-not (and network ledger type)
+                                           (throw
+                                             (ex-info
+                                               (str "Incomplete request. At least a network, db and type are required. Provided network: "
```
Comment (zonotope) [inline src/fluree/db/peer/http_api.clj:801]: Should the "db" in this error message be ledger instead?

**Review by zonotope — APPROVED** [review]
🧹

**Pull request merged**